agent/fw_support.c

```diff
@@ -52,3 +52,42 @@ void nr_fw_support_add_logging_supportability_metric(nrtxn_t* txn,
   nrm_force_add(txn->unscoped_metrics, metname, 0);
   nr_free(metname);
 }
+
+void nr_fw_support_add_package_supportability_metric(
+    nrtxn_t* txn,
+    const char* package_name,
+    const char* package_version) {
+  if (NULL == txn || NULL == package_name || NULL == package_version) {
+    return;
+  }
+
+  char* metname = NULL;
+  char major_version[4] = {0};
+
+  /* The below for loop checks if the major version of the package is more than
+   * one digit and keeps looping until a '.' is encountered or one of the
+   * conditions is met.
+   */
+  for (int i = 0; package_version[i] && i < 4; i++) {
+    if ('.' == package_version[i]) {
+      strncpy(major_version, package_version, i);
+      major_version[i] = '\0';
+      break;
+    }
+  }
+
+  if ('\0' == major_version[0] && '\0' != package_version[0]) {
+    strncpy(major_version, package_version, 1);
+    major_version[1] = '\0';
+  }
```

Comment:
This code is making some assumptions about the version format that I am unsure we can make. For example, if the version string were simply "10", my reading of this code is that we would extract "1". Is there a reason it cannot just be something like this?
```
for (int i = 0; package_version[i] && i < 4; i++) {
    if ('.' == package_version[i]) {
      break;
    }
  major_version[i] = package_version[i];
  }
```

Reply:
Thank you for catching this! Fixed: [4b61257](https://github.com/newrelic/newrelic-php-agent/pull/868/commits/4b612579ea2e1dfc9bb1e5a588cb8713e982e1c9)

Reply:
Good idea, added: [c8f52a4](https://github.com/newrelic/newrelic-php-agent/pull/868/commits/c8f52a4e074994c11b02109aa755bafdf672d4e2)